python/GafferUI/PathListingWidget.py

```diff
@@ -902,20 +909,22 @@ def updateColumnWidths( self ) :
 		self.__recalculatingColumnWidths = True
 
 		header = self.header()
-		numColumnsToResize = header.count() - 1 # leave the last section alone, as it's expandable
+		numColumnsToResize = header.count()
+		if header.stretchLastSection() :
+			numColumnsToResize -= 1 # leave the last section alone, as it's expandable
 
 		if numColumnsToResize != len( self.__columnWidthAdjustments ) :
 			# either the first time in here, or the number of columns has
 			# changed and we want to start again with the offsets.
 			self.__columnWidthAdjustments = [ 0 ] * numColumnsToResize
 
-		del self.__idealColumnWidths[:]
+		self.__idealColumnWidths = [ 0 ] * numColumnsToResize
 		for i in range( 0, numColumnsToResize ) :
+			if header.sectionResizeMode( i ) == QtWidgets.QHeaderView.ResizeMode.Interactive :
+				idealWidth = max( header.sectionSizeHint( i ), self.sizeHintForColumn( i ) )
+				self.__idealColumnWidths[i] = idealWidth
```

Comment:
It took me a while to figure out that it should be OK to allow values of `0` in `__idealColumnWidths` because `__sectionResized` shouldn't be called for non-interactive-resize sections. It still feels a bit fragile though. The existing behaviour where we clear all the `__columnWidthAdjustments` when columns are changed also feels a bit weird - if we just added a column on the end, should that really invalidate the adjustments? It feels a bit like what we want is a dictionary mapping from column to adjustment, but that could be a bit fiddly.